File: backend/src/main/kotlin/com/daybook/api/domain/model/Transaction.kt
package com.daybook.api.domain.model

import jakarta.persistence.*
import java.time.LocalDate

@Entity
@Table(name = "transactions")
class Transaction(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @Column(nullable = false)
    val date: LocalDate,

    @Column(nullable = false)
    val description: String,

    @Column(nullable = false, unique = true)
    val reference: String,

    @OneToMany(mappedBy = "transaction", cascade = [CascadeType.ALL])
    val entries: List<Entry> = emptyList()
)

File: frontend/src/types/transaction.ts
export type EntryType = 'DEBIT' | 'CREDIT'

export interface Transaction {
    id: number
    date: string
    description: string
    reference: string
    entries: Entry[]
}

export interface Entry {
    id: number
    accountId: number
    accountName: string
    amount: number
    type: EntryType
}

